src/mono/mono/metadata/assembly.c

```diff
@@ -3159,10 +3139,21 @@ mono_assembly_get_name_internal (MonoAssembly *assembly)
 /**
  * mono_register_bundled_assemblies:
  */
+MONO_DEPRECATED
 void
 mono_register_bundled_assemblies (const MonoBundledAssembly **assemblies)
 {
-	bundles = assemblies;
+	for (int i = 0; assemblies [i]; ++i) {
+		const char *id = assemblies [i]->name;
+		MonoBundledAssemblyResource *assembly_resource = mono_bundled_resources_get_assembly_resource (id);
+		if (!assembly_resource)
```

Comment:
> There is no check for adding the same key twice in the new api, but I'll add one

No additional checks should be done in the new API, we optimize that API for performance.

Reply:
I think we should consider mixing the old and new API incorrect usage. The only scenario we might need to support, since we standardize on the hash map implementation from the old API, is adding and assembly using the old API and then add symbol info for the same assembly using the old API or the reverse. If we handle that case we can always assert that the fields that should be 0 are zero in case you hit an existing item in the hash map using the same id.

Reply:
When adding a `MonoBundledAssemblyResource` either through `mono_register_bundled_assemblies` or through `mono_register_symfile_for_assembly`, asserted that all fields in the `MonoBundledAssemblyResource->assembly` or `MonoBundledAssemblyResource->symbol_data` are 0.